Frameworks/Foundation/NSURLConnection.mm

```diff
@@ -19,168 +20,333 @@
+            for (auto pair : _scheduledRunLoops) {
+                [pair.first performSelector:@selector(startLoading) target:_protocol argument:nil order:0 modes:@[ pair.second.get() ]];
+            }
+
+        } else {
+            // Run on the current run loop in the default mode
+            [[NSRunLoop currentRunLoop] performSelector:@selector(startLoading)
+                                                 target:_protocol
+                                               argument:nil
+                                                  order:0
+                                                  modes:@[ NSDefaultRunLoopMode ]];
+        }
     }
+}
 
-    [_protocol startLoading];
+/**
+ @Status Caveat
+ @Notes  TODO #2469: Cannot currently schedule on more than one runloop
+                 Callbacks do not currently occur on scheduled runloop
+*/
+- (void)scheduleInRunLoop:(NSRunLoop*)aRunLoop forMode:(NSRunLoopMode)mode {
+    @synchronized(self) {
+        if (_started) {
+            TraceError(TAG, L"[NSURLConnection scheduleInRunLoop:] called on a connection that already started.");
+            return;
+        }
+
+        if (_delegateQueue) {
+            [NSException raise:NSInternalInconsistencyException
+                        format:@"A connection cannot be scheduled with both an operation queue and a run loop."];
+        }
+
+        if (_scheduledRunLoops.size() >= 1) {
+            TraceError(TAG, L"NSURLConnection: Scheduling on more than one run loop is currently not supported.");
+            return;
+        }
+
+        _scheduledRunLoops.emplace_back(aRunLoop, mode);
+    }
 }
 
 /**
- @Status Interoperable
+ @Status Caveat
+ @Notes  TODO #2469: Callbacks do not currently occur on delegate queue
+*/
+- (void)setDelegateQueue:(NSOperationQueue*)queue {
+    @synchronized(self) {
+        if (_started) {
+            TraceError(TAG, L"[NSURLConnection setDelegateQueue:] called on a connection that already started.");
+            return;
+        }
+
+        if (_scheduledRunLoops.size() > 0) {
+            [NSException raise:NSInternalInconsistencyException
+                        format:@"A connection cannot be scheduled with both an operation queue and a run loop."];
+        }
+
+        if (_delegateQueue) {
+            TraceError(TAG, L"[NSURLConnection setDelegateQueue:] called on a connection that already had a delegate queue; ignoring.");
+            return;
+        }
+
+        _delegateQueue = queue;
+    }
+}
+
+/**
+ @Status Caveat
+ @Notes  TODO #2469: Cannot currently schedule on more than one runloop
 */
-- (void)scheduleInRunLoop:(id)runLoop forMode:(id)mode {
-    _scheduled = YES;
+- (void)unscheduleFromRunLoop:(NSRunLoop*)aRunLoop forMode:(NSRunLoopMode)mode {
+    @synchronized(self) {
+        auto it = std::find(_scheduledRunLoops.begin(), _scheduledRunLoops.end());
+        if (it != _scheduledRunLoops.end()) {
+            _scheduledRunLoops.erase(it);
```

Comment:
Nope. https://developer.apple.com/reference/foundation/nsurlconnection/1409722-unschedulefromrunloop?language=objc

"Use this method to unschedule the connection from an undesired run loop and mode before starting the connection.
You cannot reschedule a connection after it has started."